datafusion/physical-plan/src/aggregates/mod.rs

```diff
@@ -647,16 +646,16 @@ impl AggregateExec {
             return false;
         }
         // ensure there are no order by expressions
-        if self.aggr_expr().iter().any(|e| e.order_bys().is_some()) {
+        if !self.aggr_expr().iter().all(|e| e.order_bys().is_empty()) {
```

Comment:
```suggestion
          if self.aggr_expr().iter().any(|e| !e.order_bys().is_empty()) {
```

This seems more straightforward and fast (any, if one meets then early exit)